OpenXR-SDK: error compiling for Raspberry Pi armhf:  "No architecture string known!"

This is about failing to cross-compile the [OpenXR-SDK project](https://github.com/KhronosGroup/OpenXR-SDK/), please redirect me if I'm at the wrong place.

The error is
> openxr/src/common/platform_utils.hpp:75:2: error: "No architecture string known!"

From [the code](https://github.com/KhronosGroup/OpenXR-SDK/blob/main/src/common/platform_utils.hpp):

https://github.com/KhronosGroup/OpenXR-SDK/blob/288d3a7ebc1ad959f62d51da75baa3d27438c499/src/common/platform_utils.hpp#L48-L75
```c
#elif (defined(__ARM_ARCH) && __ARM_ARCH >= 7 && (defined(__ARM_PCS_VFP) || defined(__ANDROID__))) || defined(_M_ARM)
#define XR_ARCH_ABI "armv7a-vfp"
#elif defined(__ARM_ARCH_5TE__)
#define XR_ARCH_ABI "armv5te"
// ...
#else
#error "No architecture string known!"
```

For my target `__ARM_ARCH` is set to `6` and the following ARM-related macros are defined:
```c
#define __ARMEL__ 1
#define __ARM_32BIT_STATE 1
#define __ARM_ACLE 200
#define __ARM_ARCH 6
#define __ARM_ARCH_6KZ__ 1
#define __ARM_ARCH_ISA_ARM 1
#define __ARM_ARCH_ISA_THUMB 1
#define __ARM_EABI__ 1
#define __ARM_FEATURE_CLZ 1
#define __ARM_FEATURE_DSP 1
#define __ARM_FEATURE_LDREX 0x4
#define __ARM_FEATURE_QBIT 1
#define __ARM_FEATURE_SAT 1
#define __ARM_FEATURE_SIMD32 1
#define __ARM_FP 0xc
#define __ARM_FP16_ARGS 1
#define __ARM_FP16_FORMAT_IEEE 1
#define __ARM_PCS 1
#define __ARM_PCS_VFP 1
#define __ARM_SIZEOF_MINIMAL_ENUM 4
#define __ARM_SIZEOF_WCHAR_T 4
#define __ARM_VFPV2__ 1
```

--- a/changes/sdk/pr.464.gh.OpenXR-SDK-Source.md
+++ b/changes/sdk/pr.464.gh.OpenXR-SDK-Source.md
@@ -1,0 +1,4 @@
+---
+- issue.463.gh.OpenXR-SDK-Source
+---
+- loader: Fix build issue for ARMv6 architectures, and note architecture naming quirk of Raspberry Pi OS in the architecture table in the loader documentation.

--- a/specification/loader/runtime.adoc
+++ b/specification/loader/runtime.adoc
@@ -319,7 +319,7 @@ architectures and ABIs is used.
 
 |`armv5te`
   |
-  | `armel`
+  | `armel` (See <<armhf-rpios-note,note>>.)
   | 32-bit ARMv5TE architecture or compatible, little endian
 
 |`mips64`
@@ -385,6 +385,11 @@ identifier for 32-bit ARM in Android.
 While the Android ABI called `armeabi-v7a` does not use VFP PCS as the
 __system__ ABI, a definition in `openxr_platform_defines.h` enables the VFP
 PCS calling convention for all OpenXR API calls and function pointers.
+
+[[armhf-rpios-note]]
+Note: Raspberry Pi OS differs in its architecture definitions from upstream
+Debian by redefining `armhf` as ARMv6, rather than ARMv7.
+Therefore, on that operating system, `armhf` maps to ID `armv5te`.
 
 [[runtime-enumeration]]
 ==== Runtime Enumeration

--- a/src/common/platform_utils.hpp
+++ b/src/common/platform_utils.hpp
@@ -47,7 +47,7 @@
 #define XR_ARCH_ABI "aarch64"
 #elif (defined(__ARM_ARCH) && __ARM_ARCH >= 7 && (defined(__ARM_PCS_VFP) || defined(__ANDROID__))) || defined(_M_ARM)
 #define XR_ARCH_ABI "armv7a-vfp"
-#elif defined(__ARM_ARCH_5TE__)
+#elif defined(__ARM_ARCH_5TE__) || (defined(__ARM_ARCH) && __ARM_ARCH > 5)
 #define XR_ARCH_ABI "armv5te"
 #elif defined(__mips64)
 #define XR_ARCH_ABI "mips64"

